pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -295,7 +314,14 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 				// User asked for a PV that is not claimed
 				// OBSERVATION: pvc is "Pending", pv is "Available"
 				glog.V(4).Infof("synchronizing unbound PersistentVolumeClaim[%s]: volume is unbound, binding", claimToClaimKey(claim))
-				if err = ctrl.bind(volume, claim); err != nil {
+				if err = checkVolumeSatisfyClaim(volume, claim); err != nil {
```

Comment:
You carefully craft a perfect error message in `checkVolumeSatisfyClaim`, which isn't then used anywhere. It would be  nice to log it in V(4) level or make it nicer in `checkVolumeSatisfyClaim` and use it directly as the event message sent to user.

Perhaps `Can't bind the claim to volume %q, the claim requires storage class %q, while the volume has incompatible class %q`?